api/webhooks/webhooks.py

```diff
@@ -88,6 +106,12 @@ def trigger_sample_webhook(
     return _call_webhook(webhook, serializer.data)
 
 
+@backoff.on_exception(
+    backoff.expo,
+    requests.exceptions.RequestException,
+    max_tries=3,
+    jitter=backoff.full_jitter,
```

Comment:
One thing that has just cropped up in my head is that I'm not sure what the _maximum_ execution time is now for one of these tasks (assuming a reasonable response time from the webhook URL). My concern for this is that we have monitoring on each of the worker threads in the task processor. If a worker thread hasn't checked for new tasks in a given time period (defaults to 3s, but configured as 20s in our SaaS environment), that thread is deemed unhealthy and the health check will report as unhealthy. It seems perfectly reasonable that, with this logic, a single task to process a webhook could exceed 3s.

Reply:
Very interesting, I will take a look into other `jitter` options.

Reply:
I have found `max_time` under the `give_up` conditions of [Backoff](https://pypi.org/project/backoff/) project. We can set it to 2s and leave 1s as a buffer. It will solve the purpose.